security/authorization/util/util_test.go

```diff
@@ -0,0 +1,131 @@
+		expectedEvaluationError   bool
+		expr                      string
+		authzArgs                 map[string]interface{}
+	}{
+		{
+			desc:                      "1: use a policy with startsWith primitive and returns a match.",
+			expectedEvaluationOutcome: true,
+			expr:                      "request.url_path.startsWith('/pkg.service/test')",
+			authzArgs:                 map[string]interface{}{"request.url_path": "/pkg.service/test"},
+		},
+		{
+			desc:                      "2: use a policy with == primitive and returns a match.",
+			expectedEvaluationOutcome: true,
+			expr:                      "connection.uri_san_peer_certificate == 'cluster/ns/default/sa/admin'",
+			authzArgs:                 map[string]interface{}{"connection.uri_san_peer_certificate": "cluster/ns/default/sa/admin"},
+		},
+		{
+			desc:                      "3: use a policy with startsWith primitive and returns no match.",
+			expectedEvaluationOutcome: false,
+			expr:                      "request.url_path.startsWith('/pkg.service/test')",
+			authzArgs:                 map[string]interface{}{"request.url_path": "/source/pkg.service/test"},
+		},
+		{
+			desc:                      "4: use a policy with startsWith and == primitive and returns a match.",
+			expectedEvaluationOutcome: true,
+			expr:                      "request.url_path == '/pkg.service/test' && connection.uri_san_peer_certificate == 'cluster/ns/default/sa/admin'",
+			authzArgs: map[string]interface{}{"request.url_path": "/pkg.service/test",
+				"connection.uri_san_peer_certificate": "cluster/ns/default/sa/admin"},
+		},
+		{
+			desc:                 "5: use a policy that has a field not present in the environment.",
+			expectedParsingError: true,
+			expr:                 "request.source_path.startsWith('/pkg.service/test')",
+			authzArgs:            map[string]interface{}{"request.url_path": "/pkg.service/test"},
+		},
+		{
+			desc:                    "6: use an argument that has a filed not included in the environment.",
+			expectedEvaluationError: true,
+			expr:                    "request.url_path.startsWith('/pkg.service/test')",
+			authzArgs:               map[string]interface{}{"request.source_path": "/pkg.service/test"},
+		},
+	} {
+		test := test
+		t.Run(test.desc, func(t *testing.T) {
+			expected, expr, authzArgs := test.expectedEvaluationOutcome, test.expr, test.authzArgs
+			checked, err := convertStringToCheckedExpr(expr, declarations)
+			if test.expectedParsingError {
+				if err == nil {
+					t.Errorf("lack of error in conversion %v", err)
+				}
+				return
+			}
+			if err != nil {
+				t.Errorf("error in conversion %v", err)
+			}
+			ast := cel.CheckedExprToAst(checked)
+			program, err := env.Program(ast)
+			if err != nil {
+				t.Errorf("error in formatting Program %v", err)
+			}
+			got, _, gotErr := program.Eval(authzArgs)
+			if test.expectedEvaluationError {
+				if gotErr == nil {
+					t.Errorf("lack of error in evaluation %v", err)
```

Comment:
Consider using t.Fatalf to stop the test if something went wrong.

Reply:
Will Do!